projects/go-lib/src/lib/components/go-timepicker/go-timepicker.component.ts

```diff
@@ -0,0 +1,94 @@
+import { Component, ElementRef, Input, OnInit, ViewChild } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { generateId } from '../../utilities/form.utils';
+
+@Component({
+  selector: 'go-timepicker',
+  styleUrls: ['./go-timepicker.component.scss'],
+  templateUrl: './go-timepicker.component.html',
+})
+
+export class GoTimepickerComponent implements OnInit {
+  displayAbove: boolean = false;
+  displayFromRight: boolean = true;
+  timeOpen: boolean = false;
+  id: string;
+  openTimeValue: string;
+
+  @Input() key: string;
+  @Input() label: string;
+  @Input() control: FormControl;
+  @Input() hints: string[];
+  @Input() placeholder: string = '';
+  @Input() theme: string = 'light';
+  @Input() appendToContent: boolean = false;
+
+  selectedTime: string;
+
+  @ViewChild('datepickerInput', { static: true }) datepickerInput: ElementRef;
+
+  ngOnInit(): void {
+    this.id = this.key || generateId(this.label, 'timepicker');
+    this.selectedTime = this.changeTimeFormat(this.control.value);
+  }
+
+  public changeTimeFormat(timeString: string): string {
+    if (!timeString) {
+      return;
+    }
+    const H: number = +timeString.substr(0, 2);
+    const h: number | string = H % 12 || 12;
+    const hour: number | string = h > 9 ? h : '0' + h;
+    const ampm: string = H < 12 || H === 24 ? 'AM' : 'PM';
+    timeString = hour + timeString.substr(2, 3) + ' ' + ampm;
+    return timeString;
+  }
+
+  public toggleTimepicker(event: Event): void {
+    event.stopPropagation();
+    // Have to disable this here because of the event that we need to stop propagation on.
+    if (this.control.disabled) {
+      return;
+    }
+
+    this.timeOpen = !this.timeOpen;
+    const position: string = 'top';
+    const distance: object = this.datepickerInput.nativeElement.getBoundingClientRect();
+    this.displayAbove =
+      window.innerHeight - distance[position] < 350 && !this.appendToContent;
+    const convert24Hr: string = /^\s*$/.test(this.selectedTime)
+      ? undefined
+      : this.convertTo24Hour(this.selectedTime);
+    this.openTimeValue = convert24Hr ? convert24Hr : undefined;
+  }
+
+  public timePicked(event: any): void {
```

Comment:
Can we add tests for this with both `AM` and `PM` times? Just want to make sure that either way we're converting to 24H time's correctly